includes/amp-helper-functions.php

```diff
@@ -495,37 +545,37 @@ function amp_remove_endpoint( $url ) {
  * If there are known validation errors for the current URL then do not output anything.
  *
  * @since 1.0
- * @global WP_Query $wp_query
  */
 function amp_add_amphtml_link() {
-	global $wp_query;
-
 	/**
 	 * Filters whether to show the amphtml link on the frontend.
 	 *
-	 * @todo This filter's name is incorrect. It's not about adding a canonical link but adding the amphtml link.
 	 * @since 0.2
 	 */
-	if ( false === apply_filters( 'amp_frontend_show_canonical', true ) ) {
-		return;
+	if ( has_filter( 'amp_frontend_show_canonical' ) ) {
+		_deprecated_hook( 'amp_frontend_show_canonical', '1.6', 'amp_frontend_show_amphtml' );
```

Comment:
I don't think this is correct usage. You should rather just use `apply_filters_deprecated()` below.